worlds/checksmate/Options.py

```diff
@@ -0,0 +1,431 @@
+from dataclasses import dataclass
+from typing import Callable, Dict
+
+from Options import Range, Option, Choice, NamedRange, ItemDict, PerGameCommonOptions, OptionSet, DeathLink
+
+
+class Goal(Choice):
+    """
+    How victory is defined.
+
+    Single: Your opponent starts with an army of 7 pieces and 8 pawns. You have a king. Finding checkmate is your goal.
+    To get there, find checks, mate!
+
+    Ordered Progressive: When you deliver checkmate, you instead graduate to a Super-Sized board. Your goal is to
+    checkmate again, on that board!
+
+    Progressive: As Ordered Progressive, but the board grows larger when someone sends you your Super-Sized board.
+
+    Super: You skip the 8x8 board immediately. Nearly equivalent to adding Super-Size Me to your starting_inventory.
+    (Don't do that.)
```

Comment:
I think handling such an arbitrary requirement is a no-op based on the client implementation.

I still think it's fair to recommend to the user not to do that, but maybe there's a sense of professionalism in such text that you're recommending.

Let me know if you think either of these issues are unresolved.

Reply:
So the client doesn't have to handle the extra items *well*, but it does need to handle them, even if that means completely ignoring them. The big thing here is that the _logic_ needs to avoid false positives: it can't think you can do something that you can't do by assuming you can use the `start_inventory` items that you aren't actually given.

Reply:
The comment about "Nearly equivalent to..." is expressing that it does a bit of extra work but you can have roughly the same experience by doing something similar, i.e. using the start_inventory. It's... flippant, I guess.

I'm fairly sure nothing bad happens right now to the generator if you do use that start_inventory. I guess you might have an extra copy, but it should have no effect.